tests/Support/SupportArrTest.php

```diff
@@ -1533,4 +1533,32 @@ public function testSelect()
             [],
         ], Arr::select($array, null));
     }
+
+    public function testReject()
+    {
+        $array = [1, 2, 3, 4, 5, 6];
+
+        // Test rejection behavior (removing even numbers)
+        $result = Arr::reject($array, function ($value) {
+            return $value % 2 === 0;
+        });
+
+        $this->assertEquals([
+            0 => 1,
+            2 => 3,
+            4 => 5,
+        ], $result);
+
+        // Test key preservation with associative array
+        $assocArray = ['a' => 1, 'b' => 2, 'c' => 3, 'd' => 4];
+
+        $result = Arr::reject($assocArray, function ($value) {
+            return $value > 2;
+        });
+
+        $this->assertEquals([
+            'a' => 1,
+            'b' => 2,
+        ], $result);
+    }
```

Comment:
Shouldn't this ideally be two separate test case methods?
```suggestion

    public function testRejectList()
    {
        $array = [1, 2, 3, 4, 5, 6];

        // Test rejection behavior (removing even numbers)
        $result = Arr::reject($array, function ($value) {
            return $value % 2 === 0;
        });

        $this->assertEquals([
            0 => 1,
            2 => 3,
            4 => 5,
        ], $result);
    }

    public function testRejectAssociateArray()
    {
        // Test key preservation with associative array
        $assocArray = ['a' => 1, 'b' => 2, 'c' => 3, 'd' => 4];

        $result = Arr::reject($assocArray, function ($value) {
            return $value > 2;
        });

        $this->assertEquals([
            'a' => 1,
            'b' => 2,
        ], $result);
    }
```

Reply:
Yes, that’s correct, I agree with this point as well. However, my intention wasn’t directly about maintainability(updated my last message).

Each test case should indeed focus on only one thing, but what’s even more important is that the business language and style in the tests should be consistent and uniform.

In my opinion, if I were to split this into separate test cases and write it differently now, it might not be a good idea because it would differ from the style of the other tests.

Reply:
However, that begs the question how we will ever improve that if we can never really start doing so